brave/brave-core
# Propose clear data and re-join sync chain when impossible to decrypt the sync seed

Resolves https://github.com/brave/brave-browser/issues/25028

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
It is similar to the issue https://github.com/brave/brave-browser/issues/25028 STR:

1. create a sync chain via brave://settings/braveSync/setup and close the browser entirely, save the passphrase somewhere
2. go into Keychain Access and find Brave Safe Storage and remove by right clicking and selecting Delete Brave Safe Storage
3. re-launch the browser and you'll notice that you'll get the error notification implemented from 
4. Open brave://settings/braveSync/setup
Ensure you see message
`Your operating system's safe storage is unlocked, but we could not decrypt the passphrase. It looks that the current encryption key is different from the one used to encrypt the passphrase. If you have other device in the sync chain, you can reset and re-join the sync chain using the passphrase from the other device`
and the button
```[Reset and re-join sync chain]```

5. Press the button. Expected to get to `Enter the sync code` screen
6. Enter the passphrase form the pt1. Expected to join the chain.

## Screenshots:
| Description | Screenshot |
| ------------- | ------------- |
| New message instead of sync settings screen  |![Screenshot from 2022-12-02 18-53-25](https://user-images.githubusercontent.com/24739341/205347222-9ed37ef7-54de-4fa3-8785-cc7cbd2952e5.png)|
| Warning  |![Screenshot from 2022-12-02 18-53-44](https://user-images.githubusercontent.com/24739341/205347294-a7a4608c-6e6e-4c54-9d2a-188d302ab684.png)|
|Regular enter code words dialog, which is opened automatically|![Screenshot from 2022-12-02 18-54-06](https://user-images.githubusercontent.com/24739341/205347336-88aaf8ef-bbb0-4962-9163-0103fbd2d816.png)|
